src/target/riscv/riscv.c

```diff
@@ -3781,8 +3751,12 @@ static int riscv_openocd_step_impl(struct target *target, int current,
 			return ERROR_FAIL;
```

Comment:
I'm wondering if this algorithm for temporary disabling of breakpoints suffers from the same issue as the watchpoint disabling algorithm, as described here: https://github.com/riscv-collab/riscv-openocd/issues/1108

Reply:
I don't think so.
The thing is -- only one breakpoint is removed (the one that was hit) (this will probably result in issues once context-aware BPs are supported). So there is no issue of re-arrangement -- the Bp will be placed on the same trigger if the issue with WPs is fixed and the order of operations is as follows:
1. remove BP
2. remove WPs
3. restore WPs (if the issue is fixed, WPs occupy the same triggers they used to before removal)
4. restore BP